File: app/src/main/java/com/example/ruslanyussupov/popularmovies/di/AppComponent.java
package com.example.ruslanyussupov.popularmovies.di;

import com.example.ruslanyussupov.popularmovies.adapters.MovieAdapter;
import com.example.ruslanyussupov.popularmovies.data.Repository;
import com.example.ruslanyussupov.popularmovies.detail.DetailViewModel;
import com.example.ruslanyussupov.popularmovies.list.MainViewModel;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, NetworkModule.class, DataModule.class})
public interface AppComponent {

    void inject(MainViewModel mainViewModel);
    void inject(DetailViewModel detailViewModel);
    void inject(Repository repository);
    void inject(MovieAdapter movieAdapter);

}

File: app/src/main/java/com/example/ruslanyussupov/popularmovies/di/AppComponent.kt
package com.example.ruslanyussupov.popularmovies.di

import com.example.ruslanyussupov.popularmovies.data.Repository
import com.example.ruslanyussupov.popularmovies.detail.DetailViewModel
import com.example.ruslanyussupov.popularmovies.browse.MainViewModel
import com.example.ruslanyussupov.popularmovies.data.remote.MoviesRequest
import com.example.ruslanyussupov.popularmovies.data.remote.ReviewsRequest
import com.example.ruslanyussupov.popularmovies.data.remote.VideosRequest

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = [AppModule::class, NetworkModule::class, DataModule::class])
interface AppComponent {

    fun inject(mainViewModel: MainViewModel)
    fun inject(detailViewModel: DetailViewModel)
    fun inject(repository: Repository)
    fun inject(moviesRequest: MoviesRequest)
    fun inject(reviewsRequest: ReviewsRequest)
    fun inject(videosRequest: VideosRequest)

}
